// bitcore/networks.js
var Put = require('bufferput');
var buffertools = require('buffertools');
var hex = function(hex) {
  return new Buffer(hex, 'hex');
};

exports.livenet = {
  name: 'livenet',
  magic: hex('ceafcadd'),
  addressVersion: 0x24,
  privKeyVersion: 0xA4,
  P2SHVersion: 85,
  hkeyPublicVersion: 0x02cfbede,
  hkeyPrivateVersion: 0x02cfbf60,
  genesisBlock: {
    hash: hex('1def3c0d77551b1998906195b435c36318ab1ce32c547d26f7d1deea6e0d0000'),
    merkle_root: hex('1e769409f7f62f89070a12f1cade7d38bc26870a9b726c72e465eecdf0596190'),
    height: 0,
    nonce: 1650768,
    version: 1,
    prev_hash: buffertools.fill(new Buffer(32), 0),
    timestamp: 1453482000,
    bits: 504365055,
  },
  dnsSeeds: [
    '139.129.45.230',
	
  ],
  defaultClientPort: 5516
};

exports.mainnet = exports.livenet;

exports.testnet = {
  name: 'testnet',
  magic: hex('0b110907'),
  addressVersion: 0x6f,
  privKeyVersion: 239,
  P2SHVersion: 196,
  hkeyPublicVersion: 0x043587cf,
  hkeyPrivateVersion: 0x04358394,
  genesisBlock: {
    hash: hex('43497FD7F826957108F4A30FD9CEC3AEBA79972084E90EAD01EA330900000000'),
    merkle_root: hex('3BA3EDFD7A7B12B27AC72C3E67768F617FC81BC3888A51323A9FB8AA4B1E5E4A'),
    height: 0,
    nonce: 414098458,
    version: 1,
    prev_hash: buffertools.fill(new Buffer(32), 0),
    timestamp: 1296688602,
    bits: 486604799,
  },
  dnsSeeds: [
    'testnet-seed.bitcoin.petertodd.org',
    'testnet-seed.bluematt.me'
  ],
  defaultClientPort: 18333
};
